kadai1/oamam/imgconv/imgconv.go

```diff
@@ -0,0 +1,125 @@
+package imgconv
+
+import (
+	"errors"
+	"image"
+	"image/gif"
+	"image/jpeg"
+	"image/png"
+	"os"
+	"path/filepath"
+	"strings"
+)
+
+type ext string
+
+const (
+	GIF  ext = "gif"
+	PNG  ext = "png"
+	JPG  ext = "jpg"
+	JPEG ext = "jpeg"
+)
+
+var extensions = []ext{GIF, PNG, JPG, JPEG}
+
+type target struct {
+	inputPath  string
+	inputExt   ext
+	fileName   string
+	outputPath string
+	outputExt  ext
+}
+
+func (t *target) GetInputFile() string {
+	return t.inputPath + t.fileName + "." + string(t.inputExt)
+}
+
+func (t *target) GetOutputFile() string {
+	return t.outputPath + t.fileName + "." + string(t.outputExt)
+}
+
+func getTargets(id *string, od *string, ie *string, oe *string) ([]target, error) {
+	var targets []target
+	err := filepath.Walk(*id, func(path string, info os.FileInfo, err error) error {
+		if filepath.Ext(path) == "."+*ie {
+			t := target{
+				inputPath:  filepath.Dir(path) + "/",
+				inputExt:   ext(*ie),
+				fileName:   strings.TrimSuffix(filepath.Base(path), "."+*ie),
+				outputPath: strings.TrimRight(*od, "/") + "/",
+				outputExt:  ext(*oe),
+			}
+			targets = append(targets, t)
+		}
+		return nil
+	})
+	if err != nil {
+		return nil, err
+	}
+	return targets, nil
+}
+
+func decode(t target) (image.Image, error) {
```

Comment:
target構造体の固有の関数な感じがしますが、どうでしょうか。(`encode`関数もです)
```suggestion
func (t *target) decode (image.Image, error) {
```